app/controllers/jobseekers/govuk_one_login_callbacks_controller.rb

```diff
@@ -10,6 +10,11 @@ def openid_connect
         session[:user_exists_first_log_in] = { value: "true", path: "/", expires: 1.hour.from_now }
       end
 
+      if no_existing_jobseeker_with_email_exists?(govuk_one_login_user)
+        session[:newly_created_user] = { value: "true", path: "/", expires: 1.hour.from_now }
+      end
+
+
       jobseeker = Jobseeker.find_or_create_from_govuk_one_login(email: govuk_one_login_user.email,
                                                                 govuk_one_login_id: govuk_one_login_user.id)
```

Comment:
From what I see here, we are at the point of having 3 different `Jobseeker.find_by(email:  govuk_one_login_user.email)` calls happening sequentially for every connection, while at least 2 of those calls (the session setting ones) are based on "A" or "B" logic exclusive from each other.

Could we refactor this to query the jobseeker once by email (is the `email: govuk_one_login_user.email` non case sensitive by default?), and then, based on the retrieved `jobseeker` action on the session & update/create the jobseeker?